mm2src/coins/lp_coins.rs

```diff
@@ -3364,12 +3356,18 @@ pub trait MmCoin:
         coin_conf["wallet_only"].as_bool().unwrap_or(false)
     }
 
+    /// Returns the MmArc context the coin is running on top of.
+    ///
+    /// This should really never fail (return `None`), but it's here since coins hold a weak
+    /// reference to the context (TODO: why?).
+    fn get_ctx(&self) -> Option<MmArc>;
```

Comment:
> Coins can only hold a weak reference to avoid cyclic reference.

we should be fixing the cyclic reference by the parent-child relationship we have for the runtime we run things in. IMO weak ref doesn't really help when the code storing this weak ref is overly complex already, u don't know then when this weak is upgraded and for how long it is kept upgraded and when it downgrades back.
weak ref makes sense when it's feasible to verify all the code that hold these weak refs and that they *eventually* downgrade it all together at the same time. it becomes hard when tens of instances now hold these weak refs and pass it here and there.
the spawning mechanism we have is already the perfect solution for this problem, we are always sure how deep we spawn any code down this tree it will die right when we die. then we don't have to think about downgrading things and re-upgrading all the time.

> Can't MmArc be acquired from somewhere else (like passing ctx directly to the instance which owns Coin)?

This is used to access the streaming manager only right now (and not in constructing the streamers, if streamers need ctx they get it from `handle_mmrpc` anyways). Check out its concrete usage for better understanding.